### ISSUE TITLE
Runtime error handling using Duktape

### ISSUE THREAD

**Issue by Wodehouse211** [svaarala/duktape#581]
Hi svaarala,

Could you please give me an idea how to handle runtime errors on javascript such as those in onclick events and invalid parameters using duktape? Is duk_safe_call effective in handling runtime errors? duk_peval helps catch syntax errors perfectly. Am I missing something.

Thanks a lot.

**Comment by svaarala** [comment]
`duk_safe_call()` would work for both errors in the user callback and also in the C code setting up the call (e.g. looking up the callback):

- duk_safe_call() into a wrapper function
- In that wrapper function, look up the onClick handler (there are many ways to arrange that, e.g. use an index and an array of callback functions stored in the Ecmascript global object)
- Still in the wrapper function, duk_call() or duk_call_method() into the handler

Another alternative is to trust your onClick handler lookup code to not throw errors. In that case you can:

- Lookup the onClick handler (carefully avoiding errors)
- duk_pcall() or duk_pcall_method() into the handler; these are protected calls and will catch errors

Note that in general it's not 100% possible to avoid errors even when looking up the onClick handler. For example, you may run out of memory which causes an error to be thrown from pretty much any call. So I would recommend the first approach, `duk_safe_call()`.

**Comment by Wodehouse211** [comment]
yes, duk_safe_call() seems to work fine. I do have another question, I read the following link

https://github.com/svaarala/duktape/issues/253 

on reference error line number and the lineNumber property works great in the js code. However if I were to retrieve the reference error line number from the C layer just like it's done with syntax errors using duk_peval, is there a way i can achieve that?
